### ISSUE TITLE
Create a context validator for credential inputs

### ISSUE THREAD

**Issue by GuillaumeFalourd** [ZupIT/ritchie-cli#575]
**What would you like to be added**:

Thinking about Ritchie CLI use for company solutions. It would be great to add a context validator inside the config.json file.

The idea would be to add a **optional field** inside the current CREDENTIAL inputs structure to check the context in which this credential has to be set for the specific formula. 

Example:

```
"inputs": [ 
    { 
        "name": "git_user", 
        "type": "CREDENTIAL_GITHUB_USERNAME",
        "context": "PROD"
    },
    { 
        "name": "git_email", 
        "type": "CREDENTIAL_GITHUB_EMAIL",
        "context": "PROD" 
    },
    { 
        "name": "git_token", 
        "type": "CREDENTIAL_GITHUB_TOKEN",
        "context": "PROD"
    } 
]
```

If the "context" field is not defined, the value will be extract from the current user local context. But if it is set, it will be necessary to check if the specified context exists.

**Why is this needed**:

- To guaranty some formulas can only be executed on a specific context.

**Comment by ahmedkrmn** [comment]
Hi @GuillaumeFalourd,
Any ideas on how to tackle this?

**Comment by GuillaumeFalourd** [comment]
Hi @ahmedkrmn , thank you for asking 👍 

The first step would to update the Field object inside the credential.go class on https://github.com/ZupIT/ritchie-cli/tree/master/pkg/credential package, and then analise in which class the validation should be done.

We are still discussing internally about this task and its benefits, so we are are not sure what would be the best solution yet. We can discuss it here together.

**Comment by ritwikchakraborty123** [comment]
Hai I want to contribute

**Comment by ritwikchakraborty123** [comment]
I want to solve this issue

**Comment by GuillaumeFalourd** [comment]
Sure @ritwikchakraborty123 
I've assigned it to you. You're welcome to ask us anything about the solution you'll implement if needed.

**Comment by ritwikchakraborty123** [comment]
Thanks sir for being so helpful...sir I would start with running it locally...and i will embed some if else condition inside to check if value is set or not ..if set there would be some more condition checking for it's existence..

**Comment by GuillaumeFalourd** [comment]
I believe your strategy is a good way to start @ritwikchakraborty123 👍 
We don't have a Developer Guide on this repository yet (for example to explain how to run the toll locally) so if you have any issue with that, ask us, we can help as well.

**Comment by ritwikchakraborty123** [comment]
What are the required things should I install to run the codebase.. thank you

**Comment by GuillaumeFalourd** [comment]
Hi @ritwikchakraborty123 , 

You'll need to have *golang* installed as well as the *make* command.

Then use the make command at the root of the project to execute the build script on the *Makefile*: https://github.com/ZupIT/ritchie-cli/blob/master/Makefile

For example with linux: `make build-linux`

This command will create an executable file on a `dist` folder according to the OS informed.

You'll then be able to execute the binary you've created going into the OS folder (with linux it will be `/dist/linux`) to check your implementation using `./rit` as it would be the `rit` command (ex: `./rit init` or `./rit demo hello-world`)

**Comment by ritwikchakraborty123** [comment]
Sir I am a windows user...is there any point going to differ for me..

**Comment by GuillaumeFalourd** [comment]
The Makefile also create a binary for windows (`make build-windows` command).
In that case, you won't use a `./rit` command, but a `rit.exe` command instead.

**Comment by henriquemoraeszup** [comment]
Hi just dropping by to promote a little more debate.

Contexts are a free string defined by each user correct? What if I create a context `Production` on my machine and someone wrote the formula with context `PROD`? By forcing a context credential we are then declaring that the formula cannot run in any other context? But isn't this the purpose of contexts, to switch between development environments freely?

**Comment by GuillaumeFalourd** [comment]
Does it make sense to maybe configure 3 default contexts : Development / Staging / Production, and only those 3 contexts could be used in those formulas config ?
Then, allow users to configure other ones if desired for proper use.

**Comment by henriquemoraeszup** [comment]
Closing since it is solved in a different way by #632
